powermetrics: add page (#22728)
tldr-pages/tldr: pages/osx/powermetrics.md

# powermetrics

> Gather and display CPU usage, power, and interrupt wakeup statistics.
> More information: <https://keith.github.io/xcode-man-pages/powermetrics.1.html>.

- Display CPU, power, and wakeup metrics using the default 5-second sample interval:

`sudo powermetrics`

- Sample metrics every 2 seconds, taking 10 samples then exit:

> `sudo powermetrics --sample-rate {{2000}} --sample-count {{10}}`

Reviewer: document shorthands for `--sample-rate` & `--sample-count` and for other arguments too

Author: done with documenting shorthands

> `sudo powermetrics {{[-i|--sample-rate]}} {{2000}} {{[-n|--sample-count]}} {{10}}`

Reviewer 2: ```suggestion
- Sample metrics every 2000 milliseconds, taking 10 samples then exit:

`sudo powermetrics {{[-i|--sample-rate]}} 2000 {{[-n|--sample-count]}} 10`
```
Placeholders indicate user choice

Author: yes done

- Save output to a file instead of `stdout`:

`sudo powermetrics {{[-o|--output-file]}} {{path/to/output.txt}}`

- Order the process list by the specified method (default: `composite`):

> `sudo powermetrics --order {{pid|wakeups|cputime|composite}}`

Reviewer: document shorthand here too

Author: done with documenting shorthands

> `sudo powermetrics --order {{pid|wakeups|cputime|composite}}`

Reviewer 2: ```suggestion
`sudo powermetrics --order {{process_id|wakeups|cputime|composite}}`
```

Author: umm `pid` seems to be the only implementation on the man page: https://keith.github.io/xcode-man-pages/powermetrics.1.html, can you please check and let me know.

Reviewer 2: Oh sorry. I acted on autopilot. I had my mind on #22636 and didn't think that `pid` is the actual sorting argument.

> - Display output in machine-readable plist format and print a usage summary on exit:

Reviewer 2: ```suggestion
- Display output in machine-readable property list format and print a usage summary on exit:
```

Author: yup will do this

> `sudo powermetrics --format {{plist}} --show-usage-summary`

Reviewer: ```suggestion
- Display output in machine-readable plist format and print a usage summary on exit:

`sudo powermetrics {{[-f|--format]}} {{plist}} --show-usage-summary`
```

Reviewer: instead of just specifying the `plist` format itd be better to have both options available i.e `{{text|plist}}`

Author: okay, have updated this.

> `sudo powermetrics --format {{plist}} --show-usage-summary`

Reviewer 2: What is `plist`?

Author: plist stands for "Property List," and it holds XML key-value pairs. **--format** formats this into human-readable form as output.

> `sudo powermetrics {{[-f|--format]}} {{text|plist}} --show-usage-summary`

Reviewer 2: ```suggestion
`sudo powermetrics {{[-f|--format]}} plist --show-usage-summary`
```
To go against the earlier comment, since the default output is text format I don't see a reason to show both.

# powermetrics

> Gather and display CPU usage, power, and interrupt wakeup statistics on macOS.
> Requires root privileges to run.

> > More information: <https://www.unix.com/man_page/osx/1/powermetrics/>.

Reviewer 3: "on macOS" is implied, because the page is in the osx/ directory.

No need to have a note about requiring root, since the commands are prepended with `sudo`.

https://github.com/tldr-pages/tldr/blob/0cb4feb3da3cd8a52e2d27b92408c562a2c29816/contributing-guides/style-guide.md#more-information-links
> For `osx`: Apple distributes the built-in man pages [in Xcode](https://developer.apple.com/documentation/os/reading_unix_manual_pages). For commands documented there, we recommend using https://keith.github.io/xcode-man-pages/
```suggestion
> Gather and display CPU usage, power, and interrupt wakeup statistics.
> More information: <https://keith.github.io/xcode-man-pages/powermetrics.1.html>.
```

- Display CPU, power, and wakeup metrics using the default 5-second sample interval:

`sudo powermetrics`

- Sample metrics every 2 seconds, taking 10 samples then exit:

> `sudo powermetrics --sample-interval {{2000}} --sample-count {{10}}`

Reviewer 3: `--sample-interval`? There is no such option in the man page.

Author: yup will update this. I was using the man page stated in the issue: #21876
 , thanks!

- Save output to a file instead of `stdout`:

> `sudo powermetrics --output-file {{path/to/output.txt}}`

Reviewer 3: Please document the short options too, in the whole page.
https://keith.github.io/xcode-man-pages/powermetrics.1.html
> -o file, --output-file file
    Output to file instead of stdout.
```suggestion
`sudo powermetrics {{[-o|--output-file]}} {{path/to/output.txt}}`
```

- Order the process list by a specified method (`pid`, `wakeups`, `cputime`, or `composite`):

> `sudo powermetrics --order {{composite}}`

Reviewer 3: ```suggestion
- Order the process list by the specified method (default: `composite`):

`sudo powermetrics --order {{pid|wakeups|cputime|composite}}`
```

# powermetrics

> Gather and display CPU usage, power, and interrupt wakeup statistics on macOS.
> Requires root privileges to run.
> More information: <https://www.unix.com/man_page/osx/1/powermetrics/>.

- Display CPU, power, and wakeup metrics using the default 5-second sample interval:

`sudo powermetrics`

- Sample metrics every 2 seconds indefinitely:

`sudo powermetrics --sample-interval {{2000}}`

- Take a specific number of samples then exit:

> `sudo powermetrics --sample-count {{10}}`

Reviewer: I think its better combine these two examples into one. Since sample count and sample interval are related

Author: Okay noted

- Save output to a file instead of `stdout`:

`sudo powermetrics --output-file {{path/to/output.txt}}`

- Order the process list by CPU time consumed:

> `sudo powermetrics --order {{cputime}}`

Reviewer: I'd prefer having all the ordering methods here instead of just `cputime`.

Author: okay noted
